src/test/mir-opt/enum_opt.trunc.EnumSizeOpt.32bit.diff

```diff
@@ -0,0 +1,82 @@
+      let mut _2: Truncatable;             // in scope 0 at $DIR/enum_opt.rs:48:7: 48:32
+      let mut _3: [u8; 1024];              // in scope 0 at $DIR/enum_opt.rs:48:22: 48:31
+      let mut _4: Truncatable;             // in scope 0 at $DIR/enum_opt.rs:49:7: 49:32
+      let mut _5: [u8; 4096];              // in scope 0 at $DIR/enum_opt.rs:49:22: 49:31
++     let mut _6: [usize; 3];              // in scope 0 at $DIR/enum_opt.rs:48:3: 48:32
++     let mut _7: u128;                    // in scope 0 at $DIR/enum_opt.rs:48:3: 48:32
++     let mut _8: usize;                   // in scope 0 at $DIR/enum_opt.rs:48:3: 48:32
++     let mut _9: usize;                   // in scope 0 at $DIR/enum_opt.rs:48:3: 48:32
++     let mut _10: *mut Truncatable;       // in scope 0 at $DIR/enum_opt.rs:48:3: 48:32
++     let mut _11: *mut u8;                // in scope 0 at $DIR/enum_opt.rs:48:3: 48:32
++     let mut _12: *const Truncatable;     // in scope 0 at $DIR/enum_opt.rs:48:3: 48:32
++     let mut _13: *const u8;              // in scope 0 at $DIR/enum_opt.rs:48:3: 48:32
++     let mut _14: [usize; 3];             // in scope 0 at $DIR/enum_opt.rs:49:3: 49:32
++     let mut _15: u128;                   // in scope 0 at $DIR/enum_opt.rs:49:3: 49:32
++     let mut _16: usize;                  // in scope 0 at $DIR/enum_opt.rs:49:3: 49:32
++     let mut _17: usize;                  // in scope 0 at $DIR/enum_opt.rs:49:3: 49:32
++     let mut _18: *mut Truncatable;       // in scope 0 at $DIR/enum_opt.rs:49:3: 49:32
++     let mut _19: *mut u8;                // in scope 0 at $DIR/enum_opt.rs:49:3: 49:32
++     let mut _20: *const Truncatable;     // in scope 0 at $DIR/enum_opt.rs:49:3: 49:32
++     let mut _21: *const u8;              // in scope 0 at $DIR/enum_opt.rs:49:3: 49:32
+      scope 1 {
+          debug a => _1;                   // in scope 1 at $DIR/enum_opt.rs:47:7: 47:12
+      }
+  
+      bb0: {
+          StorageLive(_1);                 // scope 0 at $DIR/enum_opt.rs:47:7: 47:12
+          discriminant(_1) = 0;            // scope 0 at $DIR/enum_opt.rs:47:15: 47:29
+          StorageLive(_2);                 // scope 1 at $DIR/enum_opt.rs:48:7: 48:32
+          StorageLive(_3);                 // scope 1 at $DIR/enum_opt.rs:48:22: 48:31
+          _3 = [const 0_u8; 1024];         // scope 1 at $DIR/enum_opt.rs:48:22: 48:31
+          ((_2 as B).0: [u8; 1024]) = move _3; // scope 1 at $DIR/enum_opt.rs:48:7: 48:32
+          discriminant(_2) = 1;            // scope 1 at $DIR/enum_opt.rs:48:7: 48:32
+          StorageDead(_3);                 // scope 1 at $DIR/enum_opt.rs:48:31: 48:32
+-         _1 = move _2;                    // scope 1 at $DIR/enum_opt.rs:48:3: 48:32
++         StorageLive(_6);                 // scope 1 at $DIR/enum_opt.rs:48:3: 48:32
++         _6 = const [16_usize, 4112_usize, 0_usize]; // scope 1 at $DIR/enum_opt.rs:48:3: 48:32
```

Comment:
We don't want to transform this one. Casting discriminant to usize truncates the value and maps two discriminants into the same index, so that's not right.